src/main/java/ca/corefacility/bioinformatics/irida/processing/impl/AutomatedAnalysisFileProcessor.java

```diff
@@ -0,0 +1,189 @@
+		 */
+		if (sampleForSequencingObject != null) {
+			List<Join<Project, Sample>> projectForSample = psjRepository.getProjectForSample(
+					sampleForSequencingObject.getSubject());
+
+			//get all the projects for the sample
+			for (Join<Project, Sample> j : projectForSample) {
+
+				//get the analysis templates for that project
+				List<AnalysisSubmissionTemplate> analysisSubmissionTemplatesForProject = analysisTemplateRepository.getAnalysisSubmissionTemplatesForProject(
+						j.getSubject());
+
+				//check if the project owns this sample
+				ProjectSampleJoin psj = (ProjectSampleJoin) j;
+				boolean owner = psj.isOwner();
+
+				analysisSubmissionTemplatesForProject.forEach(t -> {
+					//adding the sample name to the template
+					String name = t.getName();
+					name = name + " - " + j.getObject()
+							.getSampleName();
+					t.setName(name);
+
+					//don't try to update the sample if this project isn't the owner.  it'll fail when it tries.
+					if (!owner) {
+						t.setUpdateSamples(false);
+					}
+				});
+
+				submissionTemplates.addAll(analysisSubmissionTemplatesForProject);
+			}
+
+		} else {
+			logger.warn("Cannot find sample for sequencing object " + object.getId()
+					+ ".  Not running automated pipelines.");
+		}
+
+		return submissionTemplates;
+	}
+
+	/**
+	 * Do the work the old Assembly and SISTR file processors used to do and assign the assembly and SISTR result back
+	 * to the {@link SequencingObject}.  This will only do something if its an Assembly or SISTR analysis and if
+	 * updateSamples is true.
+	 *
+	 * @param submission       the {@link AnalysisSubmission} to check
+	 * @param sequencingObject the {@link SequencingObject} to apply results to.
+	 */
+	private void legacyFileProcessorCompatibility(AnalysisSubmission submission, SequencingObject sequencingObject) {
+		if (submission.getUpdateSamples()) {
+			try {
+				IridaWorkflow assemblyWorkflow = workflowsService.getDefaultWorkflowByType(
+						BuiltInAnalysisTypes.ASSEMBLY_ANNOTATION);
+				IridaWorkflow sistrWorkflow = workflowsService.getDefaultWorkflowByType(
+						BuiltInAnalysisTypes.SISTR_TYPING);
+
+				UUID assemblyWorkflowWorkflowIdentifier = assemblyWorkflow.getWorkflowIdentifier();
+				UUID sistrWorkflowWorkflowIdentifier = sistrWorkflow.getWorkflowIdentifier();
+
+				if (submission.getWorkflowId()
+						.equals(assemblyWorkflowWorkflowIdentifier)) {
+					// Associate the assembly submission with the seqobject
+					sequencingObject.setAutomatedAssembly(submission);
+
+					objectRepository.save(sequencingObject);
+				} else if (submission.getWorkflowId()
+						.equals(sistrWorkflowWorkflowIdentifier)) {
+					// Associate the sistr submission with the seqobject
+					sequencingObject.setSistrTyping(submission);
+
+					objectRepository.save(sequencingObject);
+				}
+
+			} catch (IridaWorkflowNotFoundException e) {
+				logger.error("Could not associate assembly workflow with analysis " + submission.getIdentifier(), e);
```

Comment:
Can you change **assembly** to **automated**.

Reply:
Done in 223b9c5a28